lib/ansible/executor/task_executor.py

```diff
@@ -422,6 +430,9 @@ def _poll_async_result(self, result, templar):
         if int(async_result.get('finished', 0)) != 1:
             return dict(failed=True, msg="async task did not complete within the requested time")
         else:
+            # propagate no_log into the result if necessary
+            if self._play_context.no_log:
+                async_result['_ansible_no_log'] = True
             return async_result
```

Comment:
instead of these 2 this patch should take care of it:
```
diff --git a/lib/ansible/executor/task_executor.py b/lib/ansible/executor/task_executor.py
index 0e34938..3bee47a 100644
--- a/lib/ansible/executor/task_executor.py
+++ b/lib/ansible/executor/task_executor.py
@@ -419,6 +419,9 @@ class TaskExecutor:
         if self._task.notify is not None:
             result['_ansible_notify'] = self._task.notify
 
+        # preserve no_log setting
+        result["_ansible_no_log"] = self._play_context.no_log
+
         # and return
         debug("attempt loop complete, returning result")
         return result
```